### DISCUSSION TITLE
Is it possible to write a rule for the IDE warning "MemberVisibilityCanBePrivate"

### DISCUSSION THREAD

**Discussion by osquitoOfTheNorth** [detekt/detekt#3464] [Q&A] [answered]
Simply put, I've been thinking about how to write a rule for this IDE warning and I have not yet come across anything that comes close. I had a look at the source code for how the IDE warning is generated and it seems not in the spirit of quite a few of the other detekt rules I have had a look at. 

Is this something that is even possible to achieve with a detekt custom rule? Any advice or insight is appreciated.

**Comment by cortinico** [comment, marked as answer]
I think that having such a rule would be great.

> Any advice or insight is appreciated.

We discussed similar rules in https://github.com/detekt/detekt/issues/2673 https://github.com/detekt/detekt/issues/2300 but those got never developed. Perhaps we can restart from there?
